src/components/Event/Event.st.css

```diff
@@ -0,0 +1,143 @@
+    CurrentTitleColor: color(fallback(value(TitleColor), value(DefaultTitleColor)));
+    CurrentAllDayBackground: color(fallback(value(AllDayBackground), value(DefaultAllDayBackground)));
+    CurrentAllDayOpacity: fallback(value(AllDayOpacity), value(DefaultAllDayOpacity));
+    CurrentTopPadding: fallback(value(TopPadding),value(DefaultTopPadding));
+    CurrentRightPadding: fallback(value(RightPadding),value(DefaultRightPadding));
+    CurrentLeftPadding: fallback(value(LeftPadding),value(DefaultLeftPadding));
+    CurrentBottomPadding: fallback(value(BottomPadding),value(DefaultBottomPadding));
+    CurrentRTLTopPadding: fallback(value(RTLTopPadding),value(DefaultRTLTopPadding));
+    CurrentRTLRightPadding: fallback(value(RTLRightPadding),value(DefaultRTLRightPadding));
+    CurrentRTLLeftPadding: fallback(value(RTLLeftPadding),value(DefaultRTLLeftPadding));
+    CurrentRTLBottomPadding: fallback(value(RTLBottomPadding),value(DefaultRTLBottomPadding));
+}
+
+.root {
+    -st-states: isMultiday, isSelected, isRightToLeft;
+    position: relative;
+    padding: value(CurrentTopPadding) value(CurrentRightPadding) value(CurrentBottomPadding) value(CurrentLeftPadding);
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+}
+
+.root:isRightToLeft {
+    position: relative;
+    direction: rtl;
+    display: table;
+    table-layout: fixed;
+    float: right;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    padding: value(CurrentRTLTopPadding) value(CurrentRTLRightPadding) value(CurrentRTLBottomPadding) value(CurrentRTLLeftPadding);
+}
+
+.time {
+    padding-right: 2px;
+    -st-mixin: TPAText(
+        MainTextFont value(CurrentFontStyle),
+        MainTextColor value(CurrentTimeColor)
+    );
+}
+
+.title {
+  -st-mixin: TPAText(
+        MainTextFont value(CurrentFontStyle),
+        MainTextColor value(CurrentTitleColor)
+    );
+}
+
+.root:isRightToLeft .title {
+    -st-mixin: TPAText(
+        MainTextFont value(CurrentFontStyle),
+        MainTextColor value(CurrentTitleColor)
+    );
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    min-height: 1.2em;
+}
+.root:isRightToLeft .time {
+    padding-right: 0;
+    padding-left: 2px;
+    display: table-cell;
+    width: 10%;
+}
+
+.beforeMixin {
+    content: '';
+    background-color: applyOpacity(value(BeforeColor), value(BeforeOpacity));
+    position: absolute;
+    left: 0px;
+    right: 0px;
+    height: 100%;
+    margin-top: -4px;
+}
+
+.root:hover::before {
+    -st-mixin: beforeMixin;
+}
+
+.root:isSelected::before {
+    -st-mixin: beforeMixin(BeforeColor color(color-5), BeforeOpacity 0.1);
+}
+
+.root:isRightToLeft .time{
+    padding-right: 0;
+    padding-left: 2px;
+    display: table-cell;
+}
+
+.root:isMultiday {
+    background: applyOpacity(value(CurrentAllDayBackground), value(CurrentAllDayOpacity));
+}
```

Comment:
So in general, I think you can use `flex` instead of table
I think you'll get the desired outcome pretty much out of the box

Reply:
for multi day ? multi day just have a background color there is no layout here

Reply:
but regarding `multiday` if it's only the background that changes, i'm not sure we need it in the component.
I think it's a specific feature that you guys need, and that you can override the bg color from your app

Reply:
flex didn't work with the absolute position of the :before element.
I took the idea from google, but if you have a better solution it would be great.

Reply:
to drop the multiday ? it is just the background i agree, but it's part of the requirements ron gave me....